src/theme/BoxLabel.ts

```diff
@@ -0,0 +1,53 @@
+import theme from "./index"
+export type BoxLabelVariant = "solid" | "outline"
+export type BoxLabelSize = "xs" | "sm" | "md" | "lg"
+export type BoxLabelStatus = "primary" | "secondary" | "magic"
+
+type ColorMap = {
+  [key in BoxLabelStatus]: string
+}
+
+export const BoxLabel = {
+  baseStyle: ({
+    variant,
+    status,
+  }: {
+    status: BoxLabelStatus
+    variant: BoxLabelVariant
+  }) => {
+    const bgMap: ColorMap = {
+      primary: variant === "outline" ? "transparent" : "brand.50",
+      secondary: variant === "outline" ? "transparent" : "gray.50",
+      magic: variant === "outline" ? "transparent" : theme.colors.gradient[3],
```

Comment:
Styles related to variants should be moved to `variants` props built in theme IMO. Using conditional statement here is unnecessary.

See: https://chakra-ui.com/docs/styled-system/customize-theme#customizing-single-components